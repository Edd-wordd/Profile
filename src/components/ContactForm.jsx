import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { Typography, Paper, Grid, Container } from '@material-ui/core'
import TextField from '@material-ui/core/TextField'
import Radio from '@material-ui/core/Radio'
import RadioGroup from '@material-ui/core/RadioGroup'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import FormControl from '@material-ui/core/FormControl'
import FormLabel from '@material-ui/core/FormLabel'
import Button from '@material-ui/core/Button'
import Select from '@material-ui/core/Select'
import InputLabel from '@material-ui/core/InputLabel'
import MenuItem from '@material-ui/core/MenuItem'
import mainTheme from '../theme'

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundImage: `url(https://images.unsplash.com/photo-1484807352052-23338990c6c6?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2100&q=80)`,
    width: '100%',
    height: 'auto',
    backgroundSize: 'cover',
    backgroundAttachment: 'fixed',
    backgroundPosition: 'center center',
    padding: '20rem 0rem 10rem 0rem',
  },
  form: {
    '& .MuiTextField-root': {
      marginTop: theme.spacing(2),
      marginLeft: theme.spacing(3.5),
    },
  },

  formHeader: {
    borderBottom: '1.5px solid black',
    padding: '1.5rem 4.5rem',
  },
  radio: {
    paddingLeft: '3rem',
  },
  selectInputWrapper: {
    paddingLeft: '2rem',
    margin: '1.5rem 0rem',
  },
  selectInput: {
    width: '50%',
    marginTop: '.25rem',
  },
  submitBtn: {
    margin: '1.5rem 0rem',
    padding: '.5rem 4rem',
    display: 'flex',
    justify: 'center',
  },
  formPaper: {
    padding: '2rem',
  },

  heading: {
    color: mainTheme.palette.defaultLight.main,
    fontSize: '3rem',
    letterSpacing: '.15rem',
    fontWeight: 600,
  },
  paragraph: {
    color: mainTheme.palette.defaultLight.main,
    padding: '.25rem 5rem 1.5rem 0rem',
    lineHeight: '1.5rem',
    letterSpacing: '.15rem',
  },
  info: {
    width: '35%',
  },
  message: {
    padding: '1.5rem 0rem 0rem 0rem',
    color: mainTheme.palette.defaultLight.main,
    letterSpacing: '.15rem',
  },
  companyCard: {
    width: '250px',
    height: 'auto',
    margin: '0rem 5rem',
    padding: '1.5rem',
    background: 'linear-gradient(45deg,#CC95C0,#1FA2FF  , #12D8FA , #A6FFCB )',
    border: 'none',
  },
}))
const textInputs = [
  {
    required: true,
    id: 'firstName',
    label: 'First Name',
    variant: 'outlined',
    size: 'small',
    style: {
      width: '45%',
    },
  },
  {
    required: true,
    id: 'lastName',
    label: 'Last Name',
    variant: 'outlined',
    size: 'small',
    style: {
      width: '45%',
    },
  },
  {
    required: true,
    id: 'outlined-required',
    label: 'Phone Number',
    variant: 'outlined',
    size: 'small',
    style: {
      width: '45%',
    },
  },
  {
    required: true,
    id: 'outlined required',
    label: 'Email',
    variant: 'outlined',
    size: 'small',
    style: {
      width: '45%',
    },
  },
  {
    required: true,
    id: 'outlined-password-input',
    label: 'Company Name',
    autoComplete: 'current-password',
    variant: 'outlined',
    fullWidth: true,
  },
  {
    required: false,
    id: 'outlined-password-input',
    label: 'Company Website or URL',
    autoComplete: 'current-password',
    variant: 'outlined',
    fullWidth: true,
  },
  {
    required: true,
    id: 'standard-multiline-flexible',
    label: 'How Can We Help You?',
    multiline: true,
    rowsMax: 4,
    variant: 'outlined',
    fullWidth: true,
  },
]
const selectInputs = [
  {
    value: 'google search',
    label: 'Google Search',
  },
  {
    value: 'google maps',
    label: 'Google Maps',
  },
  {
    value: 'repeat client',
    label: 'Reapeat Client',
  },
  {
    value: 'online',
    label: 'Online',
  },
  {
    value: 'refferal',
    label: 'Refferal',
  },
  {
    value: 'online website',
    label: 'Online Website',
  },
  {
    value: 'other',
    label: 'Other',
  },
]
const radioInputs = [
  {
    value: 'right away',
    label: 'Right Away',
  },
  {
    value: 'one week',
    label: 'One Week',
  },
  {
    value: 'one month',
    label: 'One Month',
  },
  {
    value: 'not sure',
    label: 'Not Sure',
  },
  {
    value: 'price shopping only',
    label: 'Price Shopping Only',
  },
]

function submitForm() {
  let fname = document.getElementById('firstName').value
  console.log('Button was clicked by daddy')
  console.log(fname)
}
function ContactForm() {
  const classes = useStyles()
  const [value, setValue] = React.useState('right away')
  const [reference, setReference] = React.useState('')

  const handleSelectChange = (event) => {
    setReference(event.target.value)
  }

  const handelRadioChange = (event) => {
    setValue(event.target.value)
  }
  return (
    <div className={classes.root}>
      <Container maxWidth="xl">
        <Grid container direction="row" justify="center">
          <Grid
            container
            direction="column"
            justify="center"
            alignItems="flex-start"
            className={classes.info}
            item
            xl={6}
            lg={5}
          >
            <Typography variant="h1" className={classes.heading}>
              Need to get in Touch?
            </Typography>
            <Typography variant="h6" className={classes.message}>
              Send Us A Message
            </Typography>
            <Typography variant="body1" className={classes.paragraph}>
              Please complete the form below, click here to Book an Appointment,
              give us a call or click the message box on the lower right! Either
              way, we look forward to talking with you!
            </Typography>
            <Paper elevation={4} className={classes.companyCard}>
              <Grid
                container
                direction="column"
                justify="center"
                alignItems="flex-start"
              >
                <Typography variant="h6">Company Info</Typography>
                <Typography>Mona Tech LLC</Typography>
                <Typography>Location: Las vegas, Nevada</Typography>
                <Typography>Phone: (915) 867-1023 </Typography>
                <Typography>Email: info@monaTech.io</Typography>
              </Grid>
            </Paper>
          </Grid>
          <Grid item lg={5}>
            <Paper elevation={3} className={classes.formPaper}>
              <Typography className={classes.formHeader}>
                Please fill this form out! Let us know what we can do for you!
              </Typography>
              <form className={classes.form} noValidate autoComplete="off">
                <div>
                  <Grid
                    container
                    direction="row"
                    justifyContent="center"
                    alignItems="center"
                  >
                    {textInputs.map((value, index) => (
                      <TextField
                        key={index}
                        required={value.required}
                        id={value.id}
                        label={value.label}
                        variant={value.variant}
                        size={value.size}
                        multiline={value.multiline}
                        rowsMax={value.rowsMax}
                        style={value.style}
                        fullWidth={value.fullWidth}
                      />
                    ))}
                  </Grid>
                  <div className={classes.selectInputWrapper}>
                    <InputLabel id="select-label">
                      How did you hear about us?*
                    </InputLabel>
                    <Select
                      required
                      variant="outlined"
                      labelId="select-label"
                      id="simple-select"
                      className={classes.selectInput}
                      value={reference}
                      onChange={handleSelectChange}
                    >
                      {selectInputs.map((value, index) => (
                        <MenuItem key={index} value={value.value}>
                          {value.label}
                        </MenuItem>
                      ))}
                    </Select>
                  </div>
                </div>

                <FormControl component="fieldset" className={classes.radio}>
                  <FormLabel component="legend">
                    When are you wanting to start your project?
                  </FormLabel>
                  <RadioGroup
                    aria-label="start-date"
                    name="start-date1"
                    value={value}
                    onChange={handelRadioChange}
                  >
                    {radioInputs.map((value, index) => (
                      <FormControlLabel
                        key={index}
                        value={value.value}
                        control={<Radio />}
                        label={value.label}
                      />
                    ))}
                  </RadioGroup>
                </FormControl>
                {/* <Grid
                  container
                  direction="row"
                  justifyContent="flex-end"
                  alignItems="center"
                > */}
                <Button
                  className={classes.submitBtn}
                  size="large"
                  variant="outlined"
                  aria-label="large outlined button"
                  onClick={() => submitForm()}
                >
                  SUBMIT
                </Button>
                {/* </Grid> */}
              </form>
            </Paper>
          </Grid>
        </Grid>
      </Container>
    </div>
  )
}

export default ContactForm
