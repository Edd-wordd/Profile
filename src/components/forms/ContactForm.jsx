import React from 'react'
import {
  Typography,
  Paper,
  Grid,
  Container,
  TextField,
  Button,
  MenuItem,
  Hidden,
} from '@material-ui/core'
import { Alert, AlertTitle, Snackbar } from '@mui/material'
import { useStyles } from '../styles/forms/ContactForm.styles'
import { checkPreviousDate } from '../../utils'
import axios from 'axios'

const formFieldInputs = [
  {
    id: '1001',
    name: 'firstName',
    label: 'First Name',
    type: 'text',
    variant: 'outlined',
    helperText: ' ',
    required: true,
    size: 'small',
    InputLabelProps: {
      shrink: true,
    },
    style: {
      width: '45%',
    },
  },
  {
    id: '1002',
    name: 'lastName',
    label: 'Last Name',
    type: 'text',
    variant: 'outlined',
    helperText: ' ',
    required: true,
    InputLabelProps: {
      shrink: true,
    },
    size: 'small',
    style: {
      width: '46%',
    },
  },
  {
    id: '1003',
    name: 'phoneNumber',
    label: 'Phone Number',
    type: 'text',
    placeholder: '( ___ ) - ___ - ____',
    variant: 'outlined',
    helperText: ' ',
    required: true,
    size: 'small',
    InputLabelProps: {
      shrink: true,
    },
    style: {
      width: '45%',
    },
  },
  {
    id: '1004',
    name: 'email',
    label: 'Email',
    type: 'email',
    variant: 'outlined',
    helperText: ' ',
    required: true,
    InputLabelProps: {
      shrink: true,
    },
    size: 'small',
    style: {
      width: '46%',
    },
  },
  {
    id: '1005',
    name: 'companyName',
    label: 'Company Name or URL',
    type: 'text',
    variant: 'outlined',
    helperText: ' ',
    InputLabelProps: {
      shrink: true,
    },
    required: true,
    fullWidth: true,
  },
  {
    id: '1007',
    name: 'message',
    label: 'How can we help you?',
    variant: 'outlined',
    helperText: ' ',
    InputLabelProps: {
      shrink: true,
    },
    required: true,
    multiline: true,
    rowsMax: 4,
    fullWidth: true,
  },
  {
    id: '1008',
    name: 'startDate',
    variant: 'outlined',
    label: 'Project Start Date',
    required: false,
    fullWidth: true,
    type: 'date',
    InputLabelProps: {
      shrink: true,
    },
    inputProps: {
      min: new Date().toISOString().split('T')[0],
    },
  },
  {
    id: '1009',
    name: 'whereDidYouHearAboutUs',
    label: 'How did you hear about us?',
    variant: 'outlined',
    helperText: ' ',
    InputLabelProps: {
      shrink: true,
    },
    required: false,
    fullWidth: true,
    select: true,
  },
]

const selectInputs = [
  {
    value: 'google search',
    label: 'Google Search',
  },
  {
    value: 'repeat client',
    label: 'Repeat Client',
  },
  {
    value: 'referral',
    label: 'Referral',
  },
  {
    value: 'other',
    label: 'Other',
  },
]

function ContactForm(props) {
  const classes = useStyles(props)
  const [error, setError] = React.useState(false)
  const [alert, setAlert] = React.useState(false)
  const [value, setValue] = React.useState({
    firstName: '',
    lastName: '',
    phoneNumber: '',
    email: '',
    companyName: '',
    message: '',
    startDate: '',
    whereDidYouHearAboutUs: '',
  })

  const handleChange = (e) => {
    setValue({
      ...value,
      [e.target.name]: e.target.value,
    })
  }

  const validate = () => {
    const emailRegex =
      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
    const nameRegex = '^[a-zA-Z]{2,20}$'

    const errors = {}
    errors.firstName = value.firstName.trim().match(nameRegex) ? '' : 'Please enter your first name'
    errors.lastName = value.lastName.trim().match(nameRegex) ? '' : 'Please enter your last name'
    errors.phoneNumber = phoneCheck(value.phoneNumber) ? '' : 'Please enter valid phone number'
    // errors.email = value.email.trim().match(emailRegex) ? '' : 'Please enter valid email'
    errors.email = emailCheck(value.email) ? '' : 'Please enter valid email'
    errors.companyName = value.companyName.trim() ? '' : 'Please enter company name or url'

    errors.message = value.message ? '' : 'Please let us know how we can help you'
    errors.startDate = checkPreviousDate(value.startDate)
      ? ''
      : 'Please enter a start date for your project'
    errors.whereDidYouHearAboutUs = value.whereDidYouHearAboutUs ? '' : 'Please select a source'
    setError({ ...errors })
    return Object.values(errors).every((errValues) => errValues === '')
  }

  const handleSubmit = async (e) => {
    e.preventDefault()
    const data = new FormData(e.target)
    data.append('timeStamp', new Date())
    console.log(Object.fromEntries(data))

    const simpleData = Object.fromEntries(data)
    const phoneIsValid = await phoneCheck(simpleData.phoneNumber)

    // guard clause
    if (!validate()) return
    if (validate() && phoneIsValid) {
      setError(false)
      setValue({
        firstName: '',
        lastName: '',
        phoneNumber: '',
        email: '',
        companyName: '',
        message: '',
        startDate: '',
        whereDidYouHearAboutUs: '',
      })
      try {
        axios
          .post('/api/form', simpleData)
          .then((res) => {
            console.log(res)
            setAlert(true)
          })
          .catch((err) => {
            console.log(err)
          })
      } catch (error) {
        console.log(error)
      }
    } else if (!phoneIsValid) {
      setError({ phoneNumber: 'Please enter valid phone number' })
    }
  }

  const phoneCheck = async (phone) => {
    const phoneRegex = /^\+(?:[0-9] ?){6,14}[0-9]$/
    if (phone.trim().match(phoneRegex)) {
      const myHeaders = new Headers()
      myHeaders.append('apikey', 'GIB3A34mUj4lMeoT38zEpKyrTOV0K4OA')
      const requestOptions = {
        method: 'GET',
        redirect: 'follow',
        headers: myHeaders,
      }
      try {
        const response = await fetch(
          `https://api.apilayer.com/number_verification/validate?number=${phone}`,
          requestOptions
        )
        const data = await response.json()
        console.log(data)
        return data.valid
      } catch (error) {
        console.error(error)
        return false
      }
    }

    return false
  }

  const emailCheck = async (email) => {
    const emailRegex =
      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
    if (email.trim().match(emailRegex)) {
      const myHeaders = new Headers()
      myHeaders.append('apikey', 'GIB3A34mUj4lMeoT38zEpKyrTOV0K4OA')

      const requestOptions = {
        method: 'GET',
        redirect: 'follow',
        headers: myHeaders,
      }
      try {
        const response = await fetch(
          `https://api.apilayer.com/email_verification/check?email=${email}`,
          requestOptions
        )
        const data = await response.json()
        console.log(data)
        return data
      } catch (error) {
        console.error(error)
        return false
      }
    }
    return false
  }

  return (
    <div className={classes.root}>
      <Container maxWidth="xl">
        <Grid container direction="row" justify="center">
          <Grid
            container
            direction="column"
            justify="center"
            alignItems="flex-start"
            className={classes.leftSideInfoWrapper}
            item
            xl={6}
            lg={5}
            md={10}
            xs={12}
          >
            <Typography variant="h1" className={classes.infoHeading}>
              Need to get in Touch?
            </Typography>
            <Typography variant="h6" className={classes.infoSubTitle}>
              Send Us A Message!!!
            </Typography>
            <Typography variant="body1" className={classes.infoParagraph}>
              Whether you have a question, comment, or simply want to say hello, we're all ears.
              <br />
              <br />
              Don't hesitate to send us a message using the contact form. Our team of experts is
              always ready to assist you in any way we can. We pride ourselves in giving top-notch
              costumer service and will do everything we can to ensure your complete satisfaction.
              <br />
              <br />
              So go ahead, drop us a line! We cant wait to hear from you and see how we can help you
              achieve your software development goals.
            </Typography>
          </Grid>
          <Grid item lg={5}>
            <Paper elevation={3} className={classes.formPaper}>
              <form className={classes.form} noValidate autoComplete="off" onSubmit={handleSubmit}>
                {!alert && error && (
                  <Snackbar
                    open={Boolean(error)}
                    autoHideDuration={6000}
                    onClose={() => setError(Boolean(false))}
                  >
                    <Alert
                      onClose={() => setError(Boolean(false))}
                      variant="filled"
                      severity="error"
                    >
                      <AlertTitle>Error</AlertTitle>
                      {`You are missing ${
                        Object.values(error).filter((err) => err !== '').length
                      } ${
                        Object.values(error).filter((err) => err !== '').length > 1
                          ? 'fields! Please fill them out.'
                          : 'field! Please fill it out.'
                      }`}
                    </Alert>
                  </Snackbar>
                )}
                <div>
                  <Hidden smDown>
                    {alert && (
                      <Alert
                        onClose={() => setAlert(Boolean(false))}
                        severity="success"
                        className={classes.successAlert}
                      >
                        <AlertTitle>Success</AlertTitle>
                        Your form has been sent Successfully!
                        <strong>Thank you!</strong>
                      </Alert>
                    )}
                  </Hidden>
                  <Hidden mdUp>
                    {alert && (
                      <Snackbar
                        open={Boolean(alert)}
                        autoHideDuration={6000}
                        onClose={() => setAlert(Boolean(false))}
                      >
                        <Alert onClose={() => setAlert(Boolean(false))} severity="success">
                          <AlertTitle>Success</AlertTitle>
                          Your form has been sent Successfully!
                          <strong>Thank you!</strong>
                        </Alert>
                      </Snackbar>
                    )}
                  </Hidden>
                  <Grid container direction="row" justify="center" alignItems="center">
                    {formFieldInputs.map((values) => (
                      <TextField
                        key={values.id}
                        {...values}
                        error={!!error[values.name]}
                        helperText={error[values.name] || ' '}
                        onChange={handleChange}
                        value={value[values.name]}
                        className={classes.textField}
                        autoFocus={values.name === 'firstName'}
                      >
                        {selectInputs.map((value) => (
                          <MenuItem key={value.label} {...value}>
                            {value.value}
                          </MenuItem>
                        ))}
                      </TextField>
                    ))}
                  </Grid>
                </div>
                <Button
                  className={classes.submitBtn}
                  type="submit"
                  size="large"
                  variant="contained"
                  aria-label="large outlined button"
                  color="primary"
                  id="submit"
                  disabled={alert || error}
                >
                  SUBMIT
                </Button>
              </form>
            </Paper>
          </Grid>
        </Grid>
      </Container>
    </div>
  )
}
export default ContactForm
