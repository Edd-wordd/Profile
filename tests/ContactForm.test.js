import React from 'react'
import testComponent from './testComponent'
import ContactForm from '../src/components/forms/ContactForm'

testComponent(<ContactForm />, 'ContactForm')
