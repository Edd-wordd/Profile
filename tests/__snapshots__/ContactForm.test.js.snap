// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should render ContactForm 1`] = `"Component"`;
